app/services/notifications/NotificationService.scala

```diff
@@ -16,30 +16,42 @@
 
 package services.notifications
 
-import javax.inject.{Inject, Singleton}
+import metrics.MetricIdentifiers.notificationMetric
+import metrics.SfusMetrics
 import models.Notification
 import play.api.Logging
 import repositories.NotificationsRepository
 
+import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 import scala.xml.NodeSeq
 
 @Singleton
-class NotificationService @Inject()(repository: NotificationsRepository, notificationFactory: NotificationFactory)(implicit ec: ExecutionContext)
-    extends Logging {
+class NotificationService @Inject()(metrics: SfusMetrics, repository: NotificationsRepository, notificationFactory: NotificationFactory)(
+  implicit ec: ExecutionContext
+) extends Logging {
 
-  def parseAndSave(notificationXml: NodeSeq): Future[Either[Throwable, Unit]] = {
+  def parseAndSave(notificationXml: NodeSeq): Future[Boolean] = {
     logger.info("Notification payload: " + notificationXml)
+    val timer = metrics.startTimer(notificationMetric)
```

Comment:
Is it right to move the notification metrics timer into the service layer from the controller layer? I understand that moving it here will have little impact on the timings (as at present all the controller does is call this method) but if the controller were to be changed in future to perform some additional processing (either before or after calling this method) then that extra processing time would not be included in this metric :thinking:

Reply:
:+1: